src/TransferList/TransferList.types.ts

```diff
@@ -1,14 +1,79 @@
-export type TransferListProps = {
+export type TransferListItem =
+  | string
+  | { primaryLabel: string; secondaryLabel?: string; id: string };
+
+export type TransferListProps<T> = {
+  /**
+   * The key to be used for the filter.
+   */
+  filterKey?: (item: T) => string;
+  /**
+   * Control the transfer of the items
+   */
+  handleTransfer?: (value: string[], intent: "toTarget" | "toSource") => void;
```

Comment:
What was the design decision to move away from `selectedItems` and `onChange` to `handleTransfer` and `targetListKeys`. I find it much more confusing and it matches none of the rest of our components.